<launch>
    <arg name="pkg_name" default="toy_map_viewer" />
    <arg name="start_idx" default="20000" />
    <arg name="end_idx" default="21000" />
    <arg name="count" default="1" />
    <arg name="crop_mode" default="false" />
    <arg name="random_index" default="false" />
    <arg name="input_path" default="data/issue/global_maps/" />
    <arg name="output_path" default="data/issue/converted_bin/" />

    <arg name="crop_size" default="300.0" />

    <arg name="overlap_radius" default="0.3" />
    <arg name="linearity_tolerance" default="0.02" />

    <arg name="search_radius" default="1.5" />
    <arg name="angle_threshold_deg" default="30.0" />
    <arg name="w_dist" default="1.0" />
    <arg name="w_angle" default="5.0" />

 
    <node pkg="$(arg pkg_name)" type="map_converter_node" name="map_converter" output="screen">
        <param name="package_name" value="$(arg pkg_name)" />
        <param name="start_index" value="$(arg start_idx)" />
        <param name="end_index" value="$(arg end_idx)" />
        <param name="load_count" value="$(arg count)" />
        <param name="input_folder" value="$(arg input_path)" />
        <param name="output_folder" value="$(arg output_path)" />
        <param name="crop_mode" value="$(arg crop_mode)" />
        <param name="random_index" value="$(arg random_index)" />
        <param name="crop_size" value="$(arg crop_size)"/>
        
        <param name="overlap_radius" value="$(arg overlap_radius)"/>
        <param name="linearity_tolerance" value="$(arg linearity_tolerance)"/>
        
        <param name="search_radius" value="$(arg search_radius)"/>
        <param name="angle_threshold_deg" value="$(arg angle_threshold_deg)"/>
        <param name="w_dist" value="$(arg w_dist)"/>
        <param name="w_angle" value="$(arg w_angle)"/>
    </node>
</launch>